app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -254,11 +255,35 @@ def releases
       respond_for_index :collection => response
     end
 
+    api :PUT, "/systems/:id/content_override", N_("Set content overrides for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    param :content_override, Hash, :desc => N_("Content override parameters") do
+      param :content_label, String, :desc => N_("Label of the content"), :required => true
+      param :value, [0, 1, "default"], :desc => N_("Override to 0/1, or 'default'"), :required => true
+    end
     def content_override
-      content_override = params[:content_override]
-      @system.set_content_override(content_override[:content_label],
-                                   content_override[:name], content_override[:value]) if content_override
-      respond_for_show
+      content_override = validate_content_overrides(params[:content_override])
+      @system.set_content_override(content_override[:content_label], 'enabled', content_override[:value])
+
+      content = @system.available_content
+      response = {
+        :results => content,
+        :total => content.size,
+        :subtotal => content.size
+      }
+      respond_for_index :collection => response
+    end
+
+    api :GET, "/systems/:id/content_override", N_("Get content overrides for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    def product_content
```

Comment:
Might be nice to have a test for this action, and content_override and/or validate_content_overrides.